lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -130,12 +130,28 @@ defmodule Mix.Compilers.Elixir do
     compile =
       compile
       |> List.delete(module)
-      |> Enum.reject(&match?("elixir_" <> _, Atom.to_string(&1)))
+      |> Enum.reject(fn atom ->
```

Comment:
We can make `List.delete(module)` a part of `Enum.reject/2` check:
```elixir
Enum.reject(compile, fn
  ^module ->
    true        
  {module, _function, _arity} ->
    match?("elixir_" <> _, Atom.to_string(module))
  module ->
    match?("elixir_" <> _, Atom.to_string(module))
end)
```
Also we have two exactly the same `reject`s, I think we should define anonymous function once and use it with `compile` and `runtime`:
```elixir
skip_module? = fn ...
compile = Enum.reject(compile, skip_module?)
runtime = Enum.reject(runtime, skip_module?)
```